Authentication/LoginProviders/Google/GoogleLoginProvider.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  GoogleLoginProvider.swift
+//  Authentication
+//
+//  Created by Nahuel Gladstein on 6/28/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveSwift
+import GoogleSignIn
+import enum Result.NoError
+
+public struct GoogleLoginProviderUser: LoginProviderUser {
+    public let userId: String
+    public let idToken: String
+    public let fullName: String
+    public let givenName: String
+    public let familyName: String
+    public let email: Email?
+    
+    internal init(with user: GIDGoogleUser) {
+        userId = user.userID
+        idToken = user.authentication.idToken
+        fullName = user.profile.name
+        givenName = user.profile.givenName
+        familyName = user.profile.familyName
+        email = Email(raw: user.profile.email)
+    }
+    
+}
+
+public struct GoogleLoginProviderError: LoginProviderError {
+    
+    public let googleError: Error
+    
+    public var localizedMessage: String {
+        return googleError.localizedDescription
+    }
+    
+}
+
+public final class GoogleLoginProvider: NSObject, LoginProvider {
```

Comment:
This should have documentation as well. Of what they need. Like add the clientId to the info.plist and that stuff. Or is it somewhere else?